include/highfive/H5DataType.hpp

```diff
@@ -80,6 +87,66 @@ class AtomicType : public DataType {
 
     typedef T basic_type;
 };
+
+///
+/// \brief Create a compound HDF5 datatype
+///
+class CompoundType : public DataType {
+public:
+    ///
+    /// \brief Use for defining a sub-type of compound type
+    struct member_def {
+        member_def(std::string t_name, DataType t_base_type, size_t t_offset = 0)
+          : name(std::move(t_name))
+          , base_type(std::move(t_base_type))
+          , offset(t_offset) {}
+        std::string name;
+        DataType base_type;
+        size_t offset;
+    };
+
+    CompoundType(const CompoundType& other) = default;
+
+    CompoundType(std::vector<member_def> t_members, size_t size = 0)
+      : members(std::move(t_members)) {
+        create(size);
+    }
+
+    CompoundType(const std::initializer_list<member_def>& t_members, size_t size = 0)
```

Comment:
We are creating a copy just to be able to reuse constructor.
There should be a constructor that takes an iterable